File: src/main/java/com/moovie/backend/domain/repository/UserActivitiesRepository.java
package com.moovie.backend.domain.repository;

import com.moovie.backend.domain.model.MovieReviewDraft;
import com.moovie.backend.domain.model.UserActivity;
import com.moovie.backend.domain.model.UserActivityListing;

import java.util.List;

public interface UserActivitiesRepository {

    List<UserActivity> getUserActivities(String userId);

    UserActivityListing getFriendsActivities(int page);

    void submitReview(MovieReviewDraft draft);
}

File: src/main/kotlin/org/mobyle/domain/repository/UserActivitiesRepository.kt
package org.mobyle.domain.repository

import org.mobyle.domain.model.MovieReviewDraft
import org.mobyle.domain.model.UserActivity
import org.mobyle.model.UserActivityListing

interface UserActivitiesRepository {
    suspend fun getUserActivities(userId: String): List<UserActivity>
    suspend fun getFriendsActivities(page: Int): UserActivityListing
    suspend fun submitReview(draft: MovieReviewDraft)
}
